libs/fluid-elements/tab-group/src/lib/tab-group/tab-group.ts

```diff
@@ -136,37 +156,56 @@ export class FluidTabGroup extends LitElement {
         index = this.tabChildren.length - 1;
       }
 
+      this.tabChildren[index].tabbed = true;
+      this.tabChildren[oldIndex].tabbed = false;
       this.tabChildren[index].focus();
-      this.tabChildren[index].tabindex = 0;
-      this.tabChildren[oldIndex].tabindex = -1;
-      this.activetabid = this.tabChildren[index].tabid;
     }
   }
 
-  /** Checks whether the next tab is also disabled or not and sets the next not disabled tab as active  */
+  /** Event handler for key down events. Prevention of default scroll behavior on the SPACE key */
+  private handleKeyDown(event: KeyboardEvent): void {
+    if (event.code === SPACE) {
+      event.preventDefault();
+    }
+  }
+
+  /** Checks whether the next tab is also disabled or not and sets the next available tab as active  */
   private handleDisabled(disableTabEvent: FluidTabDisabledEvent): void {
-    if (this.activetabid === disableTabEvent.disableTab) {
-      const tabToEnable = this.tabChildren.find((tab) => !tab.disabled);
-      if (tabToEnable) {
-        tabToEnable.active = true;
-      }
+    if (this.activeTabId === disableTabEvent.tabId) {
+      this.setFirstEnabledTabActive();
     }
   }
 
   /** Handles changes in the slot. Initially sets the active item (default is first) */
   private slotchange(): void {
     this.tabChildren = Array.from(this.querySelectorAll('fluid-tab'));
-    // Initially set the first tab to active
-    if (!this.activetabid && this.tabChildren.length > 0) {
-      this.tabChildren[0].active = true;
-      this.activetabid = this.tabChildren[0].tabid;
+    // Set all tabindexes to -1 because the default is 0
+    for (const tab of this.tabChildren) {
+      tab.tabindex = -1;
+    }
+    // Set a tab to active
+    const activeTab = this.tabChildren.find((tab) => tab.active);
+    if (activeTab) {
+      activeTab.tabindex = 0;
+      this.activeTabId = activeTab.tabid;
     } else {
-      for (const tab of this.tabChildren) {
-        tab.active = tab.tabid === this.activetabid;
-      }
+      this.setFirstEnabledTabActive();
     }
-    for (const tab of this.tabChildren) {
-      tab.active = tab.tabid === this.activetabid;
+  }
+
+  /** Resets the tabindex if the user lost focus without activating the selected tab */
+  private handleBlur(event: FluidTabBlurredEvent): void {
+    // Sets the selected but not activated tabs tabindex to -1
+    const toDisableTabIndexTab = this.tabChildren.find(
+      (tab) => tab.tabid === event.tabId,
+    );
+    if (toDisableTabIndexTab) {
+      toDisableTabIndexTab.tabindex = -1;
+    }
+    // Sets the active tabs tabindex to 0
+    const toEnableTabIndexTab = this.tabChildren.find((tab) => tab.active);
+    if (toEnableTabIndexTab) {
+      toEnableTabIndexTab.tabindex = 0;
```

Comment:
Please add tests for this scenario as well.

Reply:
This is not done yet, since the blurring doesn't in a test environment. I need help with this one ^^